### PR TITLE
Improve trainer handling of learning rate scheduling and logging

### PR DESCRIPTION
**Pull request by vafl**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lostella** [comment]
When merged, this solves #555

File: src/gluonts/trainer/_base.py [hunk 1]
```diff
@@ -315,26 +323,31 @@ def loop(
                         )  # TODO: handle possible exception
 
                     if not trainer.learning_rate == curr_lr:
-                        logging.info(
+                        if best_epoch_info.epoch_no == -1:
+                            raise GluonTSUserError(
+                                "Got NaN in first epoch. Try reducing initial learning rate."
+                            )
```
Comment (lostella) [inline src/gluonts/trainer/_base.py]: I'm not a fan of this, but maybe we can improve at a later stage

Comment (vafl) [reply src/gluonts/trainer/_base.py]: I tried to save the model before the training start and then use that serialized model in this case. The challenge is that we cannot serialize the model before the first batch ran through. So this would complicate things a bit (although maybe not that much).

**Comment by DayanSiddiquiNXD** [comment]
hey are there any user docs for this? I searched in these pages on the gluonTS documentation (https://ts.gluon.ai/api/gluonts/gluonts.trainer.html, https://ts.gluon.ai/api/gluonts/gluonts.model.deepar.html, https://ts.gluon.ai/api/gluonts/gluonts.model.estimator.html) as well as the extensive examples page but couldn't find anything about using learning rate based early stopping
